services/galley/src/Galley/App.hs

```diff
@@ -148,22 +148,18 @@ validateOptions l o = do
   let settings = view optSettings o
       optFanoutLimit = fromIntegral . fromRange $ currentFanoutLimit o
   when ((isJust $ o ^. optJournal) && (settings ^. setMaxTeamSize > optFanoutLimit)) $
-    if settings ^. setMaxTeamSize > hardLimit
-      then error ("setMaxTeamSize cannot be > setMaxFanoutSize if journal is enabled and setMaxTeamSize > " ++ show hardLimit)
-      else
         Logger.warn
           l
-          ( msg $
-              val
-                "Your journaling events may have some admin user ids missing. \
-                \This is fine for testing purposes but NOT for production use!!"
+          ( msg .
+              val $
+                "Your journaling events for teams larger than " <> toByteString' optFanoutLimit <>
+                " may have some admin user ids missing. \
+                \ This is fine for testing purposes but NOT for production use!!"
```

Comment:
Sorry, why do we merge " not for production " code? Shouldn't this misconfiguration prevent galley from starting up?